src/nvme/types.h

```diff
@@ -9007,4 +9025,350 @@ struct nvme_ns_mgmt_host_sw_specified {
 };
 #endif /* SWIG */
 
+/**
+ * enum nvme_lm_cdq_fields - Controller Data Queue command fields
+ *
+ * @NVME_LM_CREATE_CDQ:					Create CDQ select option
+ * @NVME_LM_DELETE_CDQ:					Delete CDQ select option
+ * @NVME_LM_CDQ_SELECT_SHIFT:				Shift to set Select (SEL) field
+ * @NVME_LM_CDQ_SELECT_MASK:				Mask to set SEL field
+ * @NVME_LM_QUEUE_TYPE_USER_DATA_MIGRATION_QUEUE:	User Data Migration Queue type
+ * @NVME_LM_QUEUE_TYPE_SHIFT:				Shift amount to set Queue Type (QT) field
+ * @NVME_LM_QUEUE_TYPE_MASK:				Mask to set QT field
+ * @NVME_LM_CREATE_CDQ_PC:				Physically Contiguous
+ * @NVME_LM_CREATE_CDQ_CNTLID_SHIFT:			Shift amount to set CNTLID field
+ * @NVME_LM_CREATE_CDQ_CNTLID_MASK:			Mask to set CNTLID field
+ * @NVME_LM_CREATE_CDQ_CDQID_SHIFT:			Shift amount to get CDQID field from Create
+ *							response in completion dword0
+ * @NVME_LM_CREATE_CDQ_CDQID_MASK:			Mask to get CNTLID field from Create
+ *							response in completion dword0
+ * @NVME_LM_DELETE_CDQ_CDQID_SHIFT:			Shift amount to set CDQID field for deletion
+ * @NVME_LM_DELETE_CDQ_CDQID_MASK:			Mask to set CDQID field for deletion
+ */
+enum nvme_lm_cdq_fields {
+	NVME_LM_CREATE_CDQ				= 0,
+	NVME_LM_DELETE_CDQ				= 1,
+	NVME_LM_CDQ_SELECT_SHIFT			= 0,
+	NVME_LM_CDQ_SELECT_MASK				= 16,
+	NVME_LM_QUEUE_TYPE_USER_DATA_MIGRATION_QUEUE	= 0,
+	NVME_LM_QUEUE_TYPE_SHIFT			= 16,
+	NVME_LM_QUEUE_TYPE_MASK				= 0xffff,
+	NVME_LM_CREATE_CDQ_PC				= 1,
+	NVME_LM_CREATE_CDQ_CNTLID_SHIFT			= 16,
```

Comment:
Should this value not be `0`

Reply:
This is where I wasn't sure - should they be absolute bit position in the CDW, or relative to the parent field - Management Specific Operation (MOS) in this case.

I'll proceed more closely to the spec - if the fields are presented as relative to MOS then the shift and mask values will reflect that as well.